### PR TITLE
Add IT test for CSS loading order

### PR DESCRIPTION
**Pull request by tanbt**
Resolves https://github.com/vaadin/flow/issues/9529

### CODE REVIEW

File: flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        open();
+        Assert.assertEquals(
+                "Styles from addStyleSheet() should have a higher priority than from @CssImport.",
+                GREEN_RGBA,
+                $(ParagraphElement.class).id(CSS_IMPORT__PAGE_ADD_STYLESHEET)
+                        .getCssValue("color"));
+    }
+
+    @Test
+    public void lumoStyleIsOverridden() {
```
Comment (caalador) [inline flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java:67]: The test name doesn't give a clear insight to what it actually tests.

**Review by caalador — CHANGES_REQUESTED** [review]
This seems to be missing application theme position in the loading order and what it overrides (should probably override everything)

**Comment by tanbt** [comment]
I don't see any mention about testing application theme CSS loading order.
The issue says:
```
Lumo styles
(2. Parent theme)
Current theme
(4. App-specific bundled css added via @CssImport)
(5. App-specific unbundled css added via @Stylesheet
(6. Page.addStylesheet)
```
Should app theme be included in this PR?

**Comment by caalador** [comment]
Current theme should equal application theme shouldn't it. As this is about getting a test that tests what comes when in the application theme documentation

File: flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java (hunk 1 above)
Comment (tanbt) [reply flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java:67]: I actually have no idea how to test Lumo styles in a more specific way.

Comment (tanbt) [reply flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java:67]: Should I create separate views to test the priority of each pair of sources: a view to test `@CssImport` vs `@StyleSheet`, another view for  `@StyleSheet` vs `page.addStylesheet`, and so on?
There could be many combinations if doing so.

Comment (caalador) [reply flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java:67]: Perhaps you could ask Rolf or Jouni about lumo tests, also Rolf might have a vision on the expected ordering. 
Having overriding tests would be perhaps better as a bunch where each step has 2 values where one is overridden or something.

File: flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        open();
+        // @Stylesheet should override color and font-size but not margin
+        assertStylesOverride("p2", BLUE_RGBA, "18px", "1px");
+    }
+
+    @Test
+    public void appTheme_overrides_all() {
+        open();
+        assertStylesOverride("p3", YELLOW_RGBA, "20px", "2px");
+    }
```
Comment (caalador) [inline flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java:69]: Should these 3 tests just be merged to one `multipleDefinitions_correctOverrides` as opening a new browser for single asserts is quite heavy and they could just mention in the assertion fail message which part failed for which exception.

**Review by caalador — DISMISSED** [review]

File: flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java (hunk 2 above)
Comment (tanbt) [reply flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java]: Done.

File: flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                STYLESHEET_LUMO_FONT_SIZE_M,
+                executeScript(
+                        "return getComputedStyle(arguments[0]).getPropertyValue('--lumo-font-size-m')",
+                        htmlElement));
+    }
+
+    @Test
+    public void multipleDefinitions_correctOverrides() {
+        open();
+        assertStylesOverride("p1", GREEN_RGBA, "16px", "1px");
```
Comment (mshabarov) [inline flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java:54]: Is it better to split out these 3 assertions into 3 separate tests and name them accordingly for better readability/maintainability?

**Review by mshabarov — CHANGES_REQUESTED** [review]

File: flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java (hunk 3 above)
Comment (mshabarov) [reply flow-tests/test-themes/src/test/java/com/vaadin/flow/uitest/ui/theme/CssLoadingIT.java:54]: Sorry, just saw Mikael's comment about it. Ignore my comment.

**Review by mshabarov — APPROVED** [review]

**Pull request merged**